content/200-concepts/100-components/01-prisma-schema/06-relations/07-referential-actions.mdx

```diff
@@ -8,6 +8,401 @@ preview: true
+  authorId  Int
+}
+
+model User {
+  id    Int    @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+##### Result of using `NoAction`?
+
+`User`'s with posts **cannot** be deleted. The `Post`'s `id` **cannot** be changed.
+
+### SetNull
+
+- `onDelete: SetNull` The scalar field of the referenced object will be set to `NULL`.
+
+- `onUpdate: SetNull` When updating the identifier of a referenced object, the scalar fields of the referencing objects will be set to `NULL`.
+
+If one or more of the relations scalar fields are required a runtime error will be thrown.
+
+```prisma file=schema.prisma highlight=4;add
+model Post {
+  id        Int     @id @default(autoincrement())
+  title     String
+  author    User    @relation(fields: [authorId], references: [id], onDelete: SetNull, onUpdate: SetNull)
+  authorId  Int
+}
+
+model User {
+  id    Int    @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+##### Result of using `SetNull`?
+
+When deleting a `User`, the `authorId` will be set to `NULL` for all its authored posts. 
+
+When changing a `User`'s `id`, the `authorId` will be set to `NULL` for all its authored posts. 
+
+### SetDefault
+
+- `onDelete: SetDefault` The scalar field of the referencing object will be set to the fields default value.
+
+- `onUpdate: SetDefault` The scalar field of the referencing object will be set to the fields default value.
+
+These require setting a default for the relation scalar field with [`@default`](../../../../reference/api-reference/prisma-schema-reference#default). If no defaults are provided for any of the scalar fields, a runtime error will be thrown.
+
+```prisma file=schema.prisma highlight=4,5;add
+model Post {
+  id          Int     @id @default(autoincrement())
+  title       String
+  authorName  String  @default("anonymous")
+  author      User    @relation(fields: [authorId], references: [id], onDelete: SetDefault, onUpdate: SetDefault)
+  authorId    Int
+}
+
+model User {
+  id        Int    @id @default(autoincrement())
+  username  String @id
+  posts     Post[]
+}
+```
+
+##### Result of using `SetDefault`?
+
+When deleting a `User`, its existing authored posts will now be authored by a `User` with `id` "anonymous". 
+
+When the `username` of a `User` changes, its existing authored posts will now be authored by `User` with `id` "anonymous".
+
+## Behavior in <=2.25.0 or without Preview flag <inlinecode>referentialActions</inlinecode>
+
+<a id="current"></a>
+
+The following behavior applies in 2.25.0 and earlier, and since 2.26.0 _without_ the `referentialActions` preview flag.
+
+When invoking the [`delete()`](../../../../guides/database/advanced-database-tasks/cascading-deletes/postgresql#91-validating-restrict) or [`deleteAll()`](../../prisma-client/crud#delete-all-records) methods using the Prisma Client, a runtime check is performed and the deletion of records prevented if they are referencing related objects. **This prevents cascade behavior**.
```

Comment:
```suggestion
When invoking the [`delete()`](../../../../guides/database/advanced-database-tasks/cascading-deletes/postgresql#91-validating-restrict) or [`deleteAll()`](../../prisma-client/crud#delete-all-records) methods using the Prisma Client on required relations, a runtime check is performed and the deletion of records prevented if they are referencing related objects. **This prevents cascade behavior**.
```

Adding `on required relations` because I don't think this applies for optional relations.